### PR TITLE
Apply new design to Connections list

### PR DESCRIPTION
**Pull request by ConstantinKV**
issue https://github.com/saltedge/sca-authenticator-ios/issues/87  
 
Simulator Screen Shot - iPhone 11 - 2020-05-20 at 14 34 25
Simulator Screen Shot - iPhone 11 - 2020-05-20 at 14 34 04
Simulator Screen Shot - iPhone 11 - 2020-05-20 at 14 33 56

### CODE REVIEW

File: Example/Authenticator/View Controllers/AuthorizationsViewController.swift [hunk 1]
```diff
@@ -152,6 +141,17 @@ final class AuthorizationsViewController: BaseViewController {
     }
 }
 
+protocol AuthorizationsViewControllerDelegate: class {
```
Comment (baller784) [inline Example/Authenticator/View Controllers/AuthorizationsViewController.swift]: This should be on top of the class

File: Example/Authenticator/View Controllers/AuthorizationsViewController.swift [hunk 2]
```diff
@@ -152,6 +141,17 @@ final class AuthorizationsViewController: BaseViewController {
...
+    func scanQrPressed()
+    func showMainNavigationMenu()
+}
+
+private struct Layout {
```
Comment (baller784) [inline Example/Authenticator/View Controllers/AuthorizationsViewController.swift]: Also this

File: Example/Authenticator/View Models/Connections/ConnectionsListViewModel.swift [hunk 3]
```diff
@@ -46,7 +46,7 @@ class ConnectionListViewModel {
     }
 
     func count() -> Int {
-        return connections.count
+        return connections.count
```
Comment (baller784) [inline Example/Authenticator/View Models/Connections/ConnectionsListViewModel.swift]: Better turn it to `var count: Int`

File: Example/Authenticator/Views/Cells/ConnectionCell.swift [hunk 4]
```diff
@@ -22,61 +22,58 @@
...
         view.clipsToBounds = true
-        view.backgroundColor = .auth_backgroundColor
+        view.backgroundColor = .extraLightGray
         return view
     }()
-    private let connectionImageView: UIImageView = {
-        let imageView = UIImageView()
-        imageView.contentMode = .scaleAspectFit
-        return imageView
+    private let logoView: UIImageView = {
```
Comment (baller784) [inline Example/Authenticator/Views/Cells/ConnectionCell.swift]: this view is `UIImageView`. Better to call it `logoImageView`

File: Example/Authenticator/Views/Cells/ConnectionCell.swift [hunk 5]
```diff
@@ -85,26 +82,54 @@ final class ConnectionCell: UITableViewCell, Dequeuable {
     }
 }
 
+private struct Layout {
```
Comment (baller784) [inline Example/Authenticator/Views/Cells/ConnectionCell.swift]: move it to top of the class

File: Example/Authenticator/View Controllers/Connection/ConnectionsViewController.swift [hunk 6]
```diff
@@ -217,8 +218,11 @@ extension ConnectionsViewController {
 }
 
 // MARK: - Actions
-// TODO: Add logic
 private extension ConnectionsViewController {
+    @objc func close() {
+        dismiss(animated: true, completion: nil)
```
Comment (baller784) [inline Example/Authenticator/View Controllers/Connection/ConnectionsViewController.swift]: can be just `dismiss(animated: true)`

File: Example/Authenticator/View Controllers/Connection/ConnectionsViewController.swift [hunk 7]
```diff
@@ -92,16 +72,41 @@ final class ConnectionsViewController: BaseViewController {
...
+    case reconnect
+    case support
+}
+
+protocol ConnectionsViewControllerDelegate: class {
+    func selected(_ connection: Connection, action: ConnectionMenuAction?)
+    func addPressed()
+}
+
+struct ConnectionsListLayout {
```
Comment (baller784) [inline Example/Authenticator/View Controllers/Connection/ConnectionsViewController.swift:87]: `private struct Layout`

File: Example/Authenticator/View Controllers/Connection/ConnectionsViewController.swift [hunk 8]
```diff
@@ -92,16 +72,41 @@ final class ConnectionsViewController: BaseViewController {
     }
 }
 
+enum ConnectionMenuAction {
```
Comment (baller784) [inline Example/Authenticator/View Controllers/Connection/ConnectionsViewController.swift]: on top

File: Example/Authenticator/View Controllers/Connection/ConnectionsViewController.swift [hunk 9]
```diff
@@ -92,16 +72,41 @@ final class ConnectionsViewController: BaseViewController {
...
+    case delete
+    case edit
+    case reconnect
+    case support
+}
+
+protocol ConnectionsViewControllerDelegate: class {
```
Comment (baller784) [inline Example/Authenticator/View Controllers/Connection/ConnectionsViewController.swift]: on top

File: Example/Authenticator/Views/Cells/ConnectionCell.swift [hunk 10]
```diff
@@ -85,26 +82,54 @@ final class ConnectionCell: UITableViewCell, Dequeuable {
...
 // MARK: - Layout
 extension ConnectionCell: Layoutable {
     func layout() {
-        contentView.addSubviews(connectionPlaceholderView, titleLabel, descriptionLabel)
+        contentView.addSubviews(cardView)
+        cardView.addSubviews(logoPlaceholderView, titleLabel, descriptionLabel)
+        logoPlaceholderView.addSubview(logoView)
 
-        connectionPlaceholderView.addSubview(connectionImageView)
+        cardView.left(to: contentView, offset: 16.0)
```
Comment (baller784) [inline Example/Authenticator/Views/Cells/ConnectionCell.swift]: use constants from `Layout` struct

File: Example/Authenticator/View Controllers/Connection/ConnectionsViewController.swift (hunk 7 above)
Comment (ConstantinKV) [reply Example/Authenticator/View Controllers/Connection/ConnectionsViewController.swift:87]: should be public. it is used outside

**Pull request merged**